### ISSUE TITLE
x32dbg freezes when looking at disassembly of debug build of MSVC program

### ISSUE THREAD

**Issue by codecat** [x64dbg/x64dbg#1912]
I'm using the latest build (March 4 2018), on Windows 7 x64, but this also happened on a build from december. (I haven't tried any other builds.)

I built this binary from a quick MSVC project: [disastest.zip](https://github.com/x64dbg/x64dbg/files/1789359/disastest.zip)

Loading this exe in x32dbg and trying to look at the disassembly makes x32dbg freeze for me (with 0% CPU usage, so it seems like it's possibly locked somewhere or something?) It only freezes when I try looking at the disassembly of the disastest module, other modules seem to work fine.

The exe was made by using the standard "Empty project" template for C++ projects in Visual Studio 2017 (Windows SDK 10.0.16299.0), adding this code to a `main.cpp` and building it in the debug configuration on x86:

```c++
#include <cstdio>

//#include <new>

int main()
{
	printf("Creating buffer..\n");
	char* buffer = new char[0x100];

	printf("Buffer created: %p\n", buffer);

	printf("Freeing buffer..\n");
	delete[] buffer;

	printf("Done!\n");
	return 0;
}
```

It works fine if I build it as a release build. Not sure if this only happens on my machine, or if this is a duplicate bug. (It's kinda hard to search for "debug" in Github issues.)

**Comment by mrexodia** [comment]
Could you try if the same happens on the dbghelp refactor branch? https://ci.appveyor.com/project/mrexodia/x64dbg/build/1.0.433/artifacts

**Comment by codecat** [comment]
That seems to stop it from freezing, nice! Though I do notice that clicking on disastest.exe from the symbols tab makes it show not all of the module and everything is 0x00, but perhaps that's a different issue. At least the freeze is fixed and I can see the disassembly now. :+1:

**Comment by mrexodia** [comment]
I did reproduce the hang and it indeed happens inside dbghelp.dll. As a workaround you could try replacing `dbghelp.dll` with the latest version from WinDbg.

The error you saw is interesting, for me it just says it didn't load any symbols for the PDB (probably my globally registered DIA instance isn't updated yet). I think this is also the same issue as this one https://developercommunity.visualstudio.com/content/problem/4631/dia-sdk-still-doesnt-support-debugfastlink.html

**Comment by Laocoon7** [comment]
I've run into the problem of both x64/32 dbg not showing exported functions from dll's. First ran into this thread: https://github.com/x64dbg/x64dbg/issues/496. That bug was fixed a while ago, but I found this thread. So I built a release version and was able to find my exported functions. I now feel as though not finding the exported functions' parent problem is MSVC debug release (and so mentioned under this bug); however it does deserve a mention.
